src/verilog/verilog_typecheck.cpp

```diff
@@ -951,6 +955,61 @@ void verilog_typecheckt::convert_function_call_or_task_enable(
 
 /*******************************************************************\
 
+Function: verilog_typecheckt::convert_paramter_override
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void verilog_typecheckt::convert_parameter_override(
+  const verilog_parameter_overridet &module_item)
+{
+  for(auto &assignment : module_item.assignments())
+  {
+    auto lhs = to_binary_expr(assignment).lhs();
+    auto rhs = to_binary_expr(assignment).rhs();
+
+    // The lhs is a sequence of module instance names using
+    // hierarchical_identifier expressions.
+    convert_expr(lhs);
+
+    // turn into identifier
+    if(lhs.id() != ID_hierarchical_identifier)
+    {
+      error().source_location = module_item.source_location();
+      error() << "defparam expected to have a hierachical identifier" << eom;
+      throw 0;
+    }
+
+    const auto &hierarchical_identifier = to_hierarchical_identifier_expr(lhs);
+
+    // The op0 of the hierarchical identifier is the module instance,
+    // the op1 is the name of the parameter in the module.
+    if(hierarchical_identifier.op0().id() != ID_symbol)
```

Comment:
See above comment: we really shouldn't need comments to explain use of an API that we have control over.

Reply:
I'll use `module` and `item`, which is the terminology in the standard.